app/Main.hs

```diff
@@ -23,61 +33,82 @@ data Command = MoveRight
              deriving (Show, Eq)
 
 tokenize :: String -> [Command]
-tokenize [] = []
-tokenize (x:xs)
-  | x == '>' = MoveRight : tokenize xs
-  | x == '<' = MoveLeft : tokenize xs
-  | x == '+' = IncCell: tokenize xs
-  | x == '-' = DecCell : tokenize xs
-  | x == '.' = PrintCell : tokenize xs
-  | x == ',' = InputCell: tokenize xs
-  | x == '[' = OpenBracket : tokenize xs
-  | x == ']' = CloseBracket: tokenize xs
-  | otherwise = tokenize xs
+tokenize = catMaybes . fmap charToCommand
+  where
+    charToCommand c =
+      case c of
+        '>' -> Just MoveRight
+        '<' -> Just MoveLeft
+        '+' -> Just IncCell
+        '-' -> Just DecCell
+        '.' -> Just PrintCell
+        ',' -> Just InputCell
+        '[' -> Just OpenBracket
+        ']' -> Just CloseBracket
+        _ -> Nothing
 
-
--- Sets a specific cell to a value
 setCell :: (Int -> Int) -> CellState -> Int -> CellState
-setCell f cs i = take i cs ++ [if j < 0 then 255 else j] ++ drop (i + 1) cs
-    where j = f $ cs !! i
+setCell _ [] _ = []
+setCell f (c:rest) 0 = f c : rest
+setCell f (c:rest) n = c : setCell f rest (n-1)
 
 incCell :: CellState -> Int -> CellState
 incCell = setCell (+1)
 
 decCell :: CellState -> Int -> CellState
-decCell = setCell (\x -> x - 1)
-
+decCell = setCell (subtract 1)
 
 -- Parse
 -- Close bracket is different from the others as it
 -- doesn't bind itself to the parse process
 parse :: [Command] -> StateT MoveState IO [Command]
-parse [] = StateT (\x -> pure ([], x))
```

Comment:
This is exactly the definition of `pure` for `StateT` :)

https://hackage.haskell.org/package/transformers-0.5.4.0/docs/src/Control.Monad.Trans.State.Lazy.html#line-201